src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala

```diff
@@ -71,7 +71,7 @@ case class ErgoTransaction(override val inputs: IndexedSeq[Input],
     Algos.hash(ByteArrayUtils.mergeByteArrays(inputs.map(_.spendingProof.proof))).tail
 
 
-  lazy val outAssetsTry: Try[(Map[ByteArrayWrapper, Long], Int)] = ErgoTransaction.extractAssets(outputCandidates)
+  lazy val outAssetsTry: Try[(Map[ByteBuffer, Long], Int)] = ErgoBoxAssetExtractor.extractAssets(outputCandidates)
```

Comment:
What is the reason for this radical change in the key type?
 ByteArrayWrapper and ByteBuffer have completely different equality. 
ByteBuffer's equality is position sensitive, i.e. any reading from it will change the key.
Kind of scary to have such object as a key in the Map.

Reply:
uff, nice catch ... the ByteArrayWrapper is obsolete and we wanted to get rid of it so an alternative was needed.

Reply:
I will go with akka's `ByteString` if you agree

Reply:
the extractAssets method has nothing to do with akka dependency.
It actually can be moved in ergo-wallet module as utility method.
I suggest to use `Seq[Byte]` as key here and put ids there in WrappedArray.
val aiWrapped = WrappedArray.ofByte(assetId)

Reply:
@aslesarenko why not to follow https://github.com/ergoplatform/ergo/issues/1132 instead ?  With your proposal it seems we're getting more and more wrappers for fixed-size byte arrays

Reply:
@kushti, @pragmaxim  yes, but #1132 is not implemented yet. The proposed pattern can be implemented though under a different name (WrappedHash, HashBytes, HashKey, etc.) to not confuse with ModifierId.
It can be used everywhere a hash is used for better safety and performance.